sdk/servicebus/service-bus/src/serviceBusMessage.ts

```diff
@@ -983,44 +935,67 @@ export class ServiceBusMessage implements ReceivedMessage {
    *
    * @returns Promise<void>
    */
-  async deadLetter(options?: DeadLetterOptions): Promise<void> {
-    const error: AmqpError = {
-      condition: Constants.deadLetterName
-    };
-    if (options) {
-      error.info = {
-        DeadLetterReason: options.deadletterReason,
-        DeadLetterErrorDescription: options.deadLetterErrorDescription
-      };
-    }
+  async deadLetter(propertiesToModify?: DeadLetterOptions & { [key: string]: any }): Promise<void> {
     log.message(
       "[%s] Deadlettering the message with id '%s'.",
       this._context.namespace.connectionId,
       this.messageId
     );
-    if (this._context.requestResponseLockedMessages.has(this.lockToken!)) {
-      await this._context.managementClient!.updateDispositionStatus(
-        this.lockToken!,
-        DispositionStatus.suspended,
-        {
-          deadLetterReason: error.condition,
-          deadLetterDescription: error.description,
-          sessionId: this.sessionId
-        }
-      );
 
-      // Remove the message from the internal map of deferred messages
-      this._context.requestResponseLockedMessages.delete(this.lockToken!);
+    const actualPropertiesToModify: Partial<DeadLetterOptions> = {
+      ...propertiesToModify
+    };
+
+    // these two fields are handled specially and don't need to be in here.
+    delete actualPropertiesToModify.deadLetterErrorDescription;
+    delete actualPropertiesToModify.deadletterReason;
+
+    const dispositionStatusOptions: DispositionStatusOptions = {
+      propertiesToModify: actualPropertiesToModify,
+      deadLetterReason: propertiesToModify?.deadletterReason,
+      deadLetterDescription: propertiesToModify?.deadLetterErrorDescription
+    };
+    return this.settleMessage(DispositionStatus.suspended, dispositionStatusOptions);
+  }
+
+  private async settleMessage(
+    operation: DispositionStatus,
+    options?: DispositionStatusOptions
+  ): Promise<void> {
+    const dispositionType = getDispositionType(operation);
```

Comment:
Do we really need this helper method? Can we not include the disposition type as an input parameter?
The helper opens up the possibility of the return type being undefined and then we are forced to handle that with using the !

Side note: In Track 2, I want to investigate why we need both enums `DispositionStatus` and `DispositionType`.

Reply:
Delivery object expects DispositionType whereas updateDispositionStatus expects DispositionStatus.

Reply:
Actually, `DispositionType` is getting used only in error messages or in `if` statements :)

Anyway, my feedback here was to avoid having the `getDispositionType` helper and plumb through the `DispositionType` just like you are doing with `DispositionStatus` for the reasons I mentioned above. But I dont feel too strongly about it anymore

Reply:
Sure, I totally agree.

Also unrelated.. I just realized that I should also verify the message lock renewal along with backup settlement(and add some tests).